system/jlib/jstring.cpp

```diff
@@ -1056,12 +1056,24 @@ StringBuffer &replaceStringNoCase(StringBuffer & result, size_t lenSource, const
 // this method will replace all occurrences of "oldStr" with "newStr"
 StringBuffer & StringBuffer::replaceString(const char* oldStr, const char* newStr)
 {
-    if (curLen)
+    if (curLen && oldStr)
     {
+        size_t oldlen = strlen(oldStr);
+        if (oldlen > curLen)
+            return *this;
+
         StringBuffer temp;
-        size_t oldlen = oldStr ? strlen(oldStr) : 0;
         size_t newlen = newStr ? strlen(newStr) : 0;
-        ::replaceString(temp, curLen, buffer, oldlen, oldStr, newlen, newStr);
+        if (oldlen == curLen)
+        {
+            if (memcmp(buffer, oldStr, oldlen) == 0)
+                temp.append(newlen, newStr);
```

Comment:
I would think this would be more efficient to copy newStr directly into buffer, rather than going through temp.

If you make that change, then temp and the call to swap() can be moved to the same block.

Reply:
I agree it would be more efficient to copy directly and only use the temp buffer when necessary.